### PR TITLE
[ENH] Read session-specific anat data

### LINKED ISSUE #135: [ENH] Handle session-specific freesurfer data in hsvs
We'll have to make this fancier someday so it can look for session-specific freesurfers

_Originally posted by @mattcieslak in https://github.com/PennLINC/qsirecon/pull/123#discussion_r1815664336_

### LINKED ISSUE #148: [ENH] Handle session-specific anatomical data
as of https://github.com/PennLINC/qsiprep/pull/832 anatomical data can be written out to session directories. QSIRecon should be able to find them and use them

### PR DESCRIPTION
**Pull request by mattcieslak**
Closes #148 and closes #135.

QSIPrep can now write out session-specific anatomical data. This PR adapts the anatomical ingression/processing workflows to look for anatomical data first in the corresponding session directory and then at the subject level.

Changes proposed:

- Add `--session-id` parameter.
- Add `--report-output-level` parameter.
- Remove `--interactive-reports-only` parameter. THIS IS A BREAKING CHANGE.
- Look for Freesurfer derivatives organized in a session-wise manner (`sub-<label>_ses-<label>`) or a longitudinal manner (`sub-<label>_ses-<label>.long.sub-<label>`).

### CODE REVIEW

File: qsirecon/tests/utils.py [hunk 1]
```diff
@@ -24,9 +24,7 @@ def download_test_data(dset, data_dir=None):
         "singleshell_output": (
             "https://upenn.box.com/shared/static/9jhf0eo3ml6ojrlxlz6lej09ny12efgg.gz"
         ),
-        "hsvs_data": (
-            "https://upenn.box.com/shared/static/8ggsyfhldqzckh1qbywlnbm9x0tin3yr.xz"
-        ),
+        "hsvs_data": ("https://upenn.box.com/shared/static/8ggsyfhldqzckh1qbywlnbm9x0tin3yr.xz"),
```
Comment (tsalo) [inline qsirecon/tests/utils.py:27]:
```suggestion
        "hsvs_data": "https://upenn.box.com/shared/static/8ggsyfhldqzckh1qbywlnbm9x0tin3yr.xz",
```

File: qsirecon/workflows/base.py [hunk 2]
```diff
@@ -85,7 +85,6 @@ def init_single_subject_recon_wf(subject_id):
         recon_workflow_input_fields,
     )
     from ..interfaces.reports import AboutSummary, SubjectSummary
-    from ..interfaces.utils import GetUnique
```
Comment (tsalo) [inline qsirecon/workflows/base.py:88]: You can get rid of the GetUnique interface completely now, right? I don't think it's used anywhere else.

**Review by tsalo — COMMENTED** [review]
The changes make sense to me.

Should we update one or more of the test datasets to have a longitudinal structure?

**Comment by tsalo** [comment]
@araikes brought up the longitudinal Freesurfer pipeline. It would be great if we supported those data by grabbing data from subject folders named like `sub-<label>_ses-<label>.long.sub-<label>` (and probably `<subject-label>_<session-label>.long.<subject-label>`).

**Comment by tsalo** [comment]
A few notes:

1. The boilerplate is not found in the HTML reports for `--subject-report-level subject` or `session`. It's fine for the default (`root`).
2. We need to test the new output naming from QSIPrep.
3. We need to test the following scenarios:
    - longitudinal data with sessionwise anatomical processing
    - longitudinal data with first-alphabetically anatomical processing
    - longitudinal data with the Freesurfer longitudinal pipeline (would be nice, but might not be necessary)

File: qsirecon/workflows/recon/tortoise.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ def init_tortoise_estimator_wf(inputs_dict, name="tortoise_recon", qsirecon_suff
     )
     omp_nthreads = config.nipype.omp_nthreads
     desc = (
-        "TORTOISE Reconstruction\n\n: "
+        "#### TORTOISE Reconstruction\n\n"
```
Comment (mattcieslak) [inline qsirecon/workflows/recon/tortoise.py:101]: I think these are needed for the auto-rendering of these in the docs

**Review by mattcieslak — COMMENTED** [review]
I think we may need to add back the #'s to the descriptions of the scalars because of how they are rendered in the docs, but we should merge this for #155

**Pull request merged**